include/boost/geometry/strategy/cartesian/side_non_robust.hpp

```diff
@@ -0,0 +1,76 @@
+// Boost.Geometry
+
+// Copyright (c) 2020, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_NON_ROBUST_HPP
+#define BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_NON_ROBUST_HPP
+
+#include <boost/geometry/util/select_most_precise.hpp>
+#include <boost/geometry/util/select_calculation_type.hpp>
+#include <boost/geometry/util/precise_math.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace side
+{
+
+/*!
+\brief Adaptive precision predicate to check at which side of a segment a point lies:
+    left of segment (>0), right of segment (< 0), on segment (0).
+\ingroup strategies
+\tparam CalculationType \tparam_calculation
+\details This predicate determines at which side of a segment a point lies
+*/
+template
+<
+    typename CalculationType = void
+>
+struct side_non_robust
```

Comment:
So with this PR we have 3 cartesian side strategies: `side_by_triangle`, `side_non_robust` and `side_robust` with `side_robust` being the default for convex_hull.

And we also have `strategy::area::cartesian` and `strategy::area::precise_cartesian`. With the latter not defined in convex_hull umbrella strategy and probably only needed for testing.

So my questions are:

1. What's the difference between `side_by_triangle` and `side_non_robust`?

2. Why side is called robust and area called precise? Is this distinction actually important or are these names more or less arbitrary? Shouldn't we think about some consistent naming?

3. Are these robust/precise strategies complementary. I.e. is it the case that they should be used together in algorithms requiring both area and side strategies? Or can they be mixed with non-robust/non-accurate strategies?

Reply:
> So with this PR we have 3 cartesian side strategies: `side_by_triangle`, `side_non_robust` and `side_robust` with `side_robust` being the default for convex_hull.

> And we also have `strategy::area::cartesian` and `strategy::area::precise_cartesian`. With the latter not defined in convex_hull umbrella strategy and probably only needed for testing.

Actually proposed in this PR https://github.com/boostorg/geometry/pull/749 while the current one build on top of https://github.com/boostorg/geometry/pull/749
 
> So my questions are:
> 
>     1. What's the difference between `side_by_triangle` and `side_non_robust`?

`side_non_robust` is faster but probably less accurate since it does not implement any heuristic it just computes the determinant. 

>     2. Why side is called robust and area called precise? Is this distinction actually important or are these names more or less arbitrary? Shouldn't we think about some consistent naming?

I chose robust keyword for computations of predicates (yes/no result) to declare the guarantee that the result will be correct. I use precise keyword for constructions (numeric result) to declare that the result is "more precise" than the more simple computation. Of course in certain cases you can make it "more precise" (and yet slower) if you use multiprecision. We should definitely use some consistent naming. An alternative to the proposal above could be to use the keyword "robust" to whatever is more involved than straightforward floating points computation. 
 
>     3. Are these robust/precise strategies complementary. I.e. is it the case that they should be used together in algorithms requiring both area and side strategies? Or can they be mixed with non-robust/non-accurate strategies?

They can be mixed. You can compute a convex hull with robust side predicate and then compute the area with the non_robust are strategy. They are independent.

Reply:
Ok, the naming makes sense. I only worry that it may be start to be confusing at some point when there are different strategies called robust, precise or accurate. What if there are two different methods of achieving robustness or precision? Would it make sense or even be possible to use some more explicit names, e.g. indicating how the calculation is performed exactly?

Reply:
I think for now we can stick to "robust" (for predicates) and "precise" (for constructions). Note that I am also OK with "robust" for all. Then, if at some point we have two different methods of robustness (or "preciseness") for the same problem (which is a bit unlikely at least w.r.t. my current understanding of the BG development) we can think of specialized names.

@barendgehrels what do you think about this PR and the discussed issues on robustness?